### PR TITLE
Export Report v1.0 for Alpha Commit

### PR DESCRIPTION
**Pull request by JordanGS**
**Export Report v1.0 for Alpha Commit**

Committing plugin Export Report v1.0 to the alpha branch along with an updated build.xml file, json library file and .classpath file.

 1) Move package info under lisence blurb …
 2) Replaced all wildcard imports with specific ones except for SpringUtilities.java
 3) Fixed usage of MessageFormat.format, removed new Object[] {path} usage
 4) Messages displayed in the GUI are now internationalized messages from Messages.properties file
 5) Changed package name to lower case along with message properties
 6) Removed some excess commenting, some still left and is to be removed in the future
 7) Removed Double semi-colon
 8) Removed auto generated constructor
 9) Removed unnecessary comments, only kept AUTHOR and SPONSOR. Ask about JavaDoc
10) Created constant for ".dump.xml"
11) Created constant for "UTF-8"
12) Used the variable NAME to set the name during initialization.
13) Printed message to logger saying the file could not be removed, also considering a gui message but deferring that for now.
14) Changed ImageIcon error message to go to logger and not system.err.println.
15) Changed the date format "EEE, d MMM yyyy HH:mm:ss z(Z)" to a constant.
16) Removed 2 occurances of e.printStackTrace() and sent the message to the logger.
17) If Arial is unavailable then use the system default for the label.
18) Removed menu.report.html.generate and replaced it with exportreport.menu.export.label
19) Bundle the oracle license notice in the add-on.
20) Include json library as part of the add on and the alpha build path.
21) Do not touch the date at the moment.
22) Removed redundant toString calls.
23) Removed redundant variable declarations.
24) Changes some message dialog to warning dialogs  …
25) Changed JOptionPage show message to standard zap show message
26) Removed the todo file and let Git track history and change requests.

### CODE REVIEW

**Comment by JordanGS** [comment]
#357

**Comment by JordanGS** [comment]
@kingthorin updated pull request with all requested changes made and as a single commit with combined history. I hope this is what you asked for, if i made a mistake please let me know :)

Thank you for all your help!

**Comment by kingthorin** [comment]
@JordanGS thanks, sorry I didn't get to look at this over the weekend. I'll try to review tonight.

**Comment by JordanGS** [comment]
@kingthorin no rush, i accidentally closed it and wanted to link it to you :)

File: src/org/zaproxy/zap/extension/exportreport/ExtensionExportReport.java [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+			frameER.dispose();
+			frameER = null;
+		}
+	}
+
+	private ZapMenuItem getMenuExportReport()
+	{
+		if (menuExportReport == null)
+		{
+			menuExportReport = new ZapMenuItem("exportreport.menu.export.label");
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/exportreport/ExtensionExportReport.java]: Missed function call here, and the following string probably isn't necessary. Perhaps:

`ZapMenuItem(Constant.messages.getString("exportreport.menu.export.label"))`

File: src/org/zaproxy/zap/extension/exportreport/FileChooser/FileType.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+package org.zaproxy.zap.extension.exportreport.FileChooser;
+
+import javax.swing.ImageIcon;
+
+/*
+ * AUTHOR		:	GORAN SARENKAPA - JordanGS
+ * SPONSOR		:	RYERSON UNIVERSITY
+ */
+
+class FileType
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/exportreport/FileChooser/FileType.java]: Public?

File: src/org/zaproxy/zap/extension/exportreport/model/AlertItem.java [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+	@XmlElement(name = "ResponseBody")
+	public void setResponseBody(String responsebody)
+	{
+		this.responsebody = responsebody;
+	}
+	
+	// ---------------------------------------------------------------
+	
+	public List<String> getArtists()
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/exportreport/model/AlertItem.java]: Artists?

**Comment by kingthorin** [comment]
Looks good to me. Couple minor things for Fixup.

File: src/org/zaproxy/zap/extension/exportreport/ExtensionExportReport.java (hunk 1 above)
Comment (thc202) [inline src/org/zaproxy/zap/extension/exportreport/ExtensionExportReport.java]: The ZapMenuItem uses that string to read the label from the resources file.

File: src/org/zaproxy/zap/extension/exportreport/FileChooser/FileType.java (hunk 2 above)
Comment (thc202) [inline src/org/zaproxy/zap/extension/exportreport/FileChooser/FileType.java]: It depends if it's going to be used by 3rd party code or not (I don't think it needs to).

File: src/org/zaproxy/zap/extension/exportreport/ExtensionExportReport.java (hunk 1 above)
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/exportreport/ExtensionExportReport.java]: Oh, ok...wow wasn't expecting that. Cool.

Disregard.

File: src/org/zaproxy/zap/extension/exportreport/model/AlertItem.java (hunk 3 above)
Comment (JordanGS) [inline src/org/zaproxy/zap/extension/exportreport/model/AlertItem.java]: I'll check this out in an hour or two and do anther commit, i might have accidentally reverted to an older version or forgot to change the name or remove an unused method.

Did you want me to try and squash so it's one commit?

**Comment by JordanGS** [comment]
@kingthorin Cool thanks :) Did you want me to try and squash so it's one commit again?

**Comment by kingthorin** [comment]
Yes that would be best.

File: src/org/zaproxy/zap/extension/exportreport/FileChooser/FileType.java (hunk 2 above)
Comment (JordanGS) [inline src/org/zaproxy/zap/extension/exportreport/FileChooser/FileType.java]: I'm confused, did you want me to change the class to

`public class FileType` from `class FileType`, or?

File: src/org/zaproxy/zap/extension/exportreport/model/AlertItem.java (hunk 3 above)
Comment (JordanGS) [inline src/org/zaproxy/zap/extension/exportreport/model/AlertItem.java]: @kingthorin I know what this was now, it's a placeholder for a 2.0 addition i have, I kept it there as a reminder but i changed the name to **placeholder**, is it okay to keep this for now? It has no detrimental effects at the moment and only serves to help me with something i am currently working to implement (more report export options).

Comment (kingthorin) [inline src/org/zaproxy/zap/extension/exportreport/model/AlertItem.java]: Fair enough.

File: src/org/zaproxy/zap/extension/exportreport/FileChooser/FileType.java (hunk 2 above)
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/exportreport/FileChooser/FileType.java]: It's fine, don't worry about it.

I was questioning whether it should have some sort of qualifier (public, private, protected).

**Comment by JordanGS** [comment]
@kingthorin I'm going to close this, i made a mistake and merged when i shouldn't have.

**Comment by kingthorin** [comment]
No worries.

If you want to add the latest changes from alpha to your feature branch. Then basically:
1) Switch to alpha.
2) Do your fetch, merge, push to upstream (origin), etc as normal.
3) Switch to your feature branch.
4) Rebase, based on upstream/alpha.

I've kept the following for reference myself (they're not perfect, but get you on the right road):
http://eclipsesource.com/blogs/tutorials/eclipse-git-tutorial-a-rebase-workflow-with-egit/
http://stackoverflow.com/questions/21559119/how-to-resolve-conflicts-in-egit

File: src/org/zaproxy/zap/extension/exportreport/FileChooser/FileType.java (hunk 2 above)
Comment (JordanGS) [inline src/org/zaproxy/zap/extension/exportreport/FileChooser/FileType.java]: Added public

**Comment by JordanGS** [comment]
@kingthorin Closing Pull request and referencing new request: #362

**Pull request closed without merging**